Design.md

```diff
@@ -309,3 +332,33 @@ Get the platform usage of global (without query parameter), tenant (with tenant_
         * `key` : `offset` and `value` : `marker`
     1. `AssumeRoleCache` for all the `User` and `S3 Credential` APIs
         * `key` : `RoleArn` and `value` : `Credentials`
+
+## SecretKey Encryption Strategy
+
+1. Secret keys are encrypted and decrypted using a cipher algorithm and secret password provided in the `crypto.yml` config file.
+1. A `key-id` property will be provided for each `secret-password` in the `crypto.yml` config file to maintain the version of the key and will be used in the Key rotation.
+1. Meta-information will be encoded with base64 and added at the beginning of the encrypted string. This meta information will include the `key-id` of the `secret-password` used.
+1. If a `nonce` exists for the encrypted string, it will be added after the meta information in the result string.
+1. All the secret passwords in the `crypto.yml` config file are base64-encoded. These secret passwords are decoded using base64 before encrypting the secret keys.
+1. Initially, only the AES256GCM cipher algorithm is supported. (The Secretbox algorithm is suggested for future releases)
+1. An ability to change the cipher algorithm will be provided.
+1. Example `crypto.yml`
+    ```yaml
+    keys:
+      - id: 2
+        cipher: Secretbox
```

Comment:
May want to fix the example, since above it's stated only AES256GCM is supported :wink:

Reply:
Fixed in 1fce348 😃